File: app/src/main/java/wangdaye/com/geometricweather/weather/json/nominatim/NominatimLocationResult.java
package wangdaye.com.geometricweather.weather.json.nominatim;

import java.util.List;

/**
 * Nominatim location result.
 * */

public class NominatimLocationResult {

    public Integer place_id;
    public Float lat;
    public Float lon;
    public String display_name;
    public Address address;

    public static class Address {
        public String city;
        public String state_district;
        public String state;
        public String country;
        public String country_code;
    }
}

File: app/src/main/java/wangdaye/com/geometricweather/weather/json/nominatim/NominatimLocationResult.kt
package wangdaye.com.geometricweather.weather.json.nominatim

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

/**
 * Nominatim location result.
 */
@Serializable
data class NominatimLocationResult(
    @SerialName("place_id") val placeId: Int,
    val lat: Double,
    val lon: Double,
    @SerialName("display_name") val displayName: String,
    val address: NominatimLocationAddress
)
